src/modules/Game.class.js

```diff
@@ -1,68 +1,381 @@
+    if (this.getStatus() !== Game.STATUS.playing) {
+      return;
+    }
 
-  /**
-   * @returns {number}
-   */
-  getScore() {}
+    let hasMoved = false;
+
+    const newState = this.state.map((row) => {
+      const filteredRow = row.filter((cell) => cell !== 0);
+      const newRow = [];
+      let merged = false;
+
+      for (let i = 0; i < filteredRow.length; i++) {
+        if (!merged && filteredRow[i] === filteredRow[i + 1]) {
+          const mergedValue = filteredRow[i] * 2;
+
+          newRow.push(mergedValue);
+          this.score += mergedValue;
+          i++;
+          merged = true;
+          hasMoved = true;
+        } else {
+          newRow.push(filteredRow[i]);
+          merged = false;
+        }
+      }
+
+      while (newRow.length < row.length) {
+        newRow.push(0);
+      }
+
+      if (!hasMoved) {
+        hasMoved = !row.every((cell, index) => cell === newRow[index]);
+      }
+
+      return newRow;
+    });
+
+    if (hasMoved) {
+      this.state = newState;
+      this.addNumbers();
+      this.setState();
+      this.checkStatus();
+    }
+
+    return hasMoved;
+  }
+
+  moveRight() {
+    if (this.getStatus() !== Game.STATUS.playing) {
+      return;
+    }
+
+    let canMove = false;
+    const newState = this.state.map((row) => {
+      const filteredRow = row.filter((cell) => cell !== 0);
+      const newRow = [];
+
+      for (let i = filteredRow.length - 1; i >= 0; i--) {
+        if (filteredRow[i] === filteredRow[i - 1]) {
+          const mergedValue = filteredRow[i] * 2;
+
+          newRow.unshift(mergedValue);
+          this.score += mergedValue;
+          i--;
+          canMove = true;
+        } else {
+          newRow.unshift(filteredRow[i]);
+        }
+      }
+
+      while (newRow.length < row.length) {
+        newRow.unshift(0);
+      }
+
+      if (!canMove && !row.every((cell, index) => cell === newRow[index])) {
+        canMove = true;
+      }
+
+      return newRow;
+    });
+
+    if (canMove) {
+      this.state = newState;
+      this.addNumbers();
+      this.setState();
+      this.checkStatus();
+    }
+
+    return canMove;
+  }
+
+  moveUp() {
+    if (this.getStatus() !== Game.STATUS.playing) {
+      return;
+    }
+
+    let canMove = false;
+    const newState = this.state.map((row) => [...row]);
+
+    for (let c = 0; c < this.state[0].length; c++) {
+      const filteredColumn = [];
+
+      for (let r = 0; r < this.state.length; r++) {
+        if (this.state[r][c] !== 0) {
+          filteredColumn.push(this.state[r][c]);
+        }
+      }
+
+      const newColumn = [];
+
+      for (let i = 0; i < filteredColumn.length; i++) {
+        if (filteredColumn[i] === filteredColumn[i + 1]) {
+          const mergedValue = filteredColumn[i] * 2;
+
+          newColumn.push(mergedValue);
+          this.score += mergedValue;
+          i++;
+          canMove = true;
+        } else {
+          newColumn.push(filteredColumn[i]);
+        }
```

Comment:
This similar logic is duplicated in every method, try to move it to a helper function and use it everywhere.